app/jobs/challenge_email_job.rb

```diff
@@ -0,0 +1,11 @@
+class ChallengeEmailJob < ApplicationJob
```

Comment:
You can do something like an `EmailJob` that looks like this if you want to only have one:

```ruby
class EmailJob < ApplicationJob
  queue_as :default

  def perform(mailer_klass:, email_method:, args:)
    mailer_klass.constantize.send(email_method, args.deep_symbolize_keys)
  end
end
```

Then call it like this:

```ruby
EmailJob.perform_later(
  mailer_klass: ChallengeSuccessMailer, 
  email_method: :challenge_success_email, 
  args: { ... }
)
```

Reply:
I like this. Thanks!